### ISSUE TITLE
Broken functionality of visualstudiogenerator + tools.collect_libs

### ISSUE THREAD

**Issue by kucerad** [conan-io/conan#5303]
The functionality of tools.collect_libs with recent changes in VisualStudioGenerator is breaking down for libs that contain "." in the name. (My.Library.lib) The tools.collect_libs() returns a list of library names (My.Library), then when in generator you try to findout if the name provided contains extension or not... so we get ["My", "Library"]  - so it seems there is an extension and you put the name as is to the props file ("MyLibrary")

https://github.com/conan-io/conan/blob/9f6d07feadd8b37e960098acb741b036fd66177f/conans/client/generators/visualstudio.py#L74

I suggest to add the name with extension here:
https://github.com/conan-io/conan/blob/develop/conans/client/tools/files.py#L332

**Comment by jgsogo** [comment]
Hi @kucerad. The retrieved name from the `tools.collect_libs` call should not contain the extension, as it is recommended if you write explicitly the library name: `self.cpp_info.libs = ["My.Library",...]`.

So this issue should be handled in the generator.

Thanks for reporting.

**Comment by kucerad** [comment]
@jgsogo I don't agree! I'd say it would be better to keep the extension if provided and guess it if not provided. What if for some ill reason I'd like to have my static libraries to have the .xyz extension? Would Conan be able to handle that?

**Comment by jgsogo** [comment]
The `self.cpp_info.libs` is intended to be the same for all the platforms (and we should think about `tools.collect_libs()` just as a helper), so the preferred way is to name the library without the extension, otherwise we would need many if-clauses like:

```python
def package_info(self):
    if self.settings.os == "Windows":
        self.cpp_info.libs = ["mylib.lib"]
    else:
        self.cpp_info.libs = ["mylib.a"]
```

It is the same reason why we don't include the `lib` prefix. This is what you will find in most of the recipes in conan-community.

I agree with you and maybe it is a problem: custom extensions or custom prefixes won't work with Conan, but I can't think right now about any issue related to it.

**Comment by jgsogo** [comment]
@kucerad  I wasn't aware of this PR (https://github.com/conan-io/conan/pull/5254), but there we have probably broken previous behavior. So the analysis on this issue could change once the other one is fixed. 
Thanks for your patience

**Comment by jgsogo** [comment]
Hi, @kucerad, we have already merged #5319 into `1.16.1` release. With this change Conan will manage some known extensions like `.lib` or `.a`(remove in the `collect_libs` and add again in the generator). There shouldn't be a problem with library names like `My.Library.lib`.

**Comment by memsharded** [comment]
VisualStudioGenerator  deprecated in favor of ``MSBuildDeps`` and ``MSBuildToolchain`` generators, closing this ticket as outdated
